sw/device/lib/testing/csrng_testutils.c

```diff
@@ -17,6 +18,58 @@ enum {
   kNumOutputWordsMax = 16,
 };
 
+/**
+ * Generates randomized seed material.
+ */
+status_t csrng_testutils_seed_material_build(
+    bool disable_rand, dif_csrng_seed_material_t *seed_material) {
+  if (seed_material == NULL) {
+    return INVALID_ARGUMENT();
+  }
+  seed_material->seed_material_len =
+      disable_rand
+          ? 0
+          : rand_testutils_gen32_range(0, kDifCsrngSeedMaterialMaxWordLen);
+  for (size_t i = 0; i < seed_material->seed_material_len; ++i) {
+    seed_material->seed_material[i] = rand_testutils_gen32();
+  }
+  return OK_STATUS();
+}
+
+/**
+ * Returns a randomized CSRNG application command.
+ */
+csrng_app_cmd_t csrng_testutils_app_cmd_build(
+    bool disable_rand, csrng_app_cmd_id_t id,
+    dif_csrng_entropy_src_toggle_t entropy_src_en, unsigned int glen_val,
+    dif_csrng_seed_material_t *seed_material) {
+  CHECK_STATUS_OK(
+      csrng_testutils_seed_material_build(disable_rand, seed_material));
+
+  // If disable_rand is true, we set flag0 to the provided value.
+  // If disable_rand is false, we pick a value at random.
+  dif_csrng_entropy_src_toggle_t entropy_src_enable =
+      disable_rand                       ? entropy_src_en
+      : rand_testutils_gen32_range(0, 1) ? kDifCsrngEntropySrcToggleEnable
```

Comment:
One thing to note here is that if for some reason this triggers an Ibex EDN0 request, we may end up in a lockup state. 

https://github.com/lowRISC/opentitan/blob/76b1c7498c102e9cdf2fc3f7fc5db2b229e8be27/sw/device/lib/testing/rand_testutils.c#L51-L57

Something you can consider is getting all the ranges needed before stopping the entropy complex, or we need a version of `rand_test_utils_gen32_range()` that won't trigger an lfsr reseed.

Reply:
Thanks @moidx for your review. The assumption here is that the entropy complex is running while we call this function, similar to 
https://github.com/lowRISC/opentitan/blob/76b1c7498c102e9cdf2fc3f7fc5db2b229e8be27/sw/device/lib/testing/edn_testutils.c#L29-L31

And in the function where we call this one, we have a `entropy_testutils_auto_mode_init();` at the top. However, this auto_mode_init call is non-ideal because it always disables and re-enables the entropy complex. Meaning it again takes a long time to get the first seed from ENTROPY_SRC (takes several ms) to actually compute these parameters. Ideally, we would infer whether the entropy complex is already running in auto mode. If we can keep the ENTROPY_SRC running throughout the test, it can fill up it's seed buffer leading to less serialization due to ENTROPY_SRC and thus more pressure on the CSRNG interfaces.

Reply:
Thanks @vogelpi sounds good!